### PR TITLE
Add Binance authentication for User Data Streams

### PR DESCRIPTION
**Pull request by Dynamikey**
This PR introduces Binance authentication for User Data Streams for Binance (SPOT), Binance Delivery (COIN-M) and Binance Futures (USD-M).

To subscribe successfully to User Data Streams, we create a listenkey that needs to be periodically refreshed in order to keep-alive. Currently, Binance supports BALANCE updates, while Binance Delivery and Binance Futures support both BALANCE and POSITION updates.

- [x] - Tested
- [x] - Changelog updated
- [x] - Tests run and pass
- [x] - Flake8 run and all errors/warnings resolved
- [ ] - Contributors file updated (optional)

### CODE REVIEW

**Review by bmoscon — CHANGES_REQUESTED** [review]

**Comment by bmoscon** [comment]
overall I don't have a major problem with this. I'd prefer if the auth mechanism didnt involve a class - that part seems unnecessary. I'd also prefer if all the auth code was just backed into the main exchange class (so in this case, in class Binance). The rest components could use the same code from the websocket class. It seems like you just need to provide two methods, the initial authentication and the periodic token refresh. I like the idea of having the start method extended in the child class and creating the task that does this periodic refresh. The stop and/or shutdown methods might similarly need to be extended to destroy/cancel this task if it exists. Let me know what you think about this. I want to remove the `auth` module, and this would be the first step in doing that

**Comment by bmoscon** [comment]
@Dynamikey sorry for all the churn but there are a few merge conflicts now :( Think you'll be able to have a look at this anytime soon?

**Comment by Dynamikey** [comment]
> @Dynamikey sorry for all the churn but there are a few merge conflicts now :( Think you'll be able to have a look at this anytime soon?

Hey Bryant! I've been preoccupried, sorry. I hope to get to it asap!

File: cryptofeed/auth/binance.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import asyncio
+import requests
+from urllib.parse import urlencode
+
+
+class BinanceAuth():
+    api = 'https://api.binance.com/api/v3/'
+    listen_key_endpoint = 'userDataStream'
```
[the remark this answers is not in the document]
Comment (Dynamikey) [reply cryptofeed/auth/binance.py:8]: What do you mean by input to the init function? The listen_key_endpoint is different for the subclasses.

[the remark this answers is not in the document]
Comment (bmoscon) [reply cryptofeed/auth/binance.py:8]: if we are going away from the class approach i think the issue is moot

**Comment by bmoscon** [comment]
@Dynamikey - I'm fine with either way. If there is some overlap with the rest code, it would be nice if that could be removed and kept in one place. It looks like there are two methods that we'd need in the binance module, the initial auth call and the refresh call. I think you can just move those methods into the binance class and in the init methods of the binance exchange objects set their auth URLs (or move that as a class member since it wont change per instance). You can in the init create a task to do the periodic refresh.

**Comment by bmoscon** [comment]
@Dynamikey I'm going to be doing a release soon, so let me know if you have a rough ETA on this.

**Review by bmoscon — APPROVED** [review]

**Comment by bmoscon** [comment]
@Dynamikey - thanks!

**Pull request merged**